packages/core/assistant-testing/src/functions/discord/fetch-messages.ts

```diff
@@ -140,3 +202,44 @@ export default defineFunction({
     Effect.orDie,
   ),
 });
+
+/**
+ * @param unixTimestamp in seconds
+ */
+const generateSnowflake = (unixTimestamp: number): bigint => {
+  const discordEpoch = 1420070400000n; // Discord Epoch (ms)
+  return (BigInt(unixTimestamp * 1000) - discordEpoch) << 22n;
+};
+
+const parseSnowflake = (snowflake: string): Date => {
+  const discordEpoch = 1420070400000n; // Discord Epoch (ms)
+  return new Date(Number((BigInt(snowflake) >> 22n) + discordEpoch));
+};
+
+const makeMessage = (message: MessageResponse): DataType.Message =>
+  Obj.make(DataType.Message, {
+    [Obj.Meta]: {
+      keys: [
+        { id: message.id, source: 'discord.com' },
+        { id: message.channel_id, source: 'discord.com/thread' },
```

Comment:
thread vs channel? it seems like you're not using channels here?

Reply:
If I understand their API correctly -- messages are always stored in channels. Channels could be "Channels" from UX like `#general` or they could be threads created by users -- see the code above that combines threads and channels.

Reply:
Fetching messages from a channel will not include messages in subthreads of that channel -- they have their own channel_id equal to thread.id